src/librustc/ty/layout.rs

```diff
@@ -1513,6 +1603,30 @@ impl<'a, 'gcx, 'tcx> Layout {
             }
         }
     }
+
+    /// Returns alignment before repr alignment is applied
+    pub fn abi_align(&self, dl: &TargetDataLayout) -> Align {
```

Comment:
Maybe `content_align`? Or, given that it seems to be recursive, `base_align`? (user-specified alignment is still ABI alignment, in the sense that it affects padding, whereas preferred alignment is for allocations, e.g. to speed up SSE).